packages/vue/src/generators/stories/lib/component-story.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  generateFiles,
+  getProjects,
+  joinPathFragments,
+  normalizePath,
+  Tree,
+} from '@nx/devkit';
+import { ensureTypescript } from '@nx/js/src/utils/typescript/ensure-typescript';
+import { StorybookStoriesSchema } from '../stories';
+import {
+  camelCase,
+  createDefautPropsObject,
+  getDefinePropsObject,
+} from './utils';
+
+let tsModule: typeof import('typescript');
+
+export function createComponentStories(
+  host: Tree,
+  { project, js, interactionTests }: StorybookStoriesSchema,
+  componentPath: string
+) {
+  if (!tsModule) {
+    tsModule = ensureTypescript();
+  }
+  const proj = getProjects(host).get(project);
+  const sourceRoot = proj.sourceRoot;
+
+  const componentFilePath = joinPathFragments(sourceRoot, componentPath);
+
+  const componentDirectory = componentFilePath.replace(
```

Comment:
Both `componentDirectory` and `componentFileName` can be computed with `parse` from `node:path`.

Like

```
const { name: componentFileName, dir: componentDirectory } = parse(componentFilePath);
```